client/state/terms/schema.js

```diff
@@ -29,3 +29,23 @@ export const termsSchema = {
 	},
 	additionalProperties: false
 };
+
+export const queriesSchema = {
+	type: 'object',
+	patternProperties: {
+		'^\\d+$': {
+			type: 'object',
+			patternProperties: {
+				// Queries are JSON strings, prepended by a site ID and taxonomy slug
+				'^([A-Za-z0-9-_]+:)\\{[^\\}]*\\}$': {
```

Comment:
I did think about breaking this out further by taxonomy type during the refactor yesterday.  I suppose it could help prevent keys from getting very very long, at the cost of a more complex schema.  I could go either way on it, what do you think is the best route?